DataFormats/CaloRecHit/interface/CaloRecHit.h

```diff
@@ -6,15 +6,13 @@
 
 #ifdef __CUDA_ARCH__
```

Comment:
```suggestion
#if defined(__CUDA_ARCH__) or defined(__HIP_DEVICE_COMPILE__)
```

to make it work also for AMD GPUs

Reply:
Alternatively, you could `#include "FWCore/Utilities/interface/HostDeviceConstant.h"` and replace
```c++
#ifdef __CUDA_ARCH__
__constant__
#endif
    constexpr uint32_t calo_rechit_masks[] = { ... };
```
with
```c++
    HOST_DEVICE_CONSTANT uint32_t calo_rechit_masks[] = { ... };
```

However, this would use global device memory instead of constant memory, and I don't know if it makes a difference in terms of performance.